src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -483,6 +483,9 @@ public async Task<SignedHttpRequestValidationResult> ValidateSignedHttpRequestAs
                 // validate signed http request signature
                 signedHttpRequest.SigningKey = await ValidateSignatureAsync(signedHttpRequest, popKey, signedHttpRequestValidationContext, cancellationToken).ConfigureAwait(false);
 
+                // validate nonce claim
+                _ = ValidateNonceAsync(signedHttpRequest, popKey, signedHttpRequestValidationContext, cancellationToken);
```

Comment:
if this returns false throw a generic SignedHttpRequestInvalidNonceClaimException, See what we do for the Validator delegates in TokenValidationParameters

Reply:
This doesn't call a local method it should call the delegate on the validation parameters.